### PR TITLE
New Feature: Specify hosted checkout form locale

### PR DESCRIPTION
**Pull request by abdulmueid**
This PR allows setting the locale for the hosted checkout form from Payment Settings in adminhtml.

Screenshot from 2020-07-28 14-54-52

Screenshot from 2020-07-28 14-55-27

### CODE REVIEW

**Comment by abdulmueid** [comment]
@dareenzo can you please test and verify it's working as expected?

**Comment by dareenzo** [comment]
@abdulmueid  It worked wonders. I set it to French just for kicks.

PS: Screenshot only looks weird because I was trying it under Incognito Mode with many restrictions set

Screenshot from 2020-07-28 19-10-13

**Comment by dareenzo** [comment]
Tested in production mode and with `setup:static-content:deploy`

Screenshot from 2020-07-30 10-31-04

**Comment by dareenzo** [comment]
Hi @OnTapKristjan

did you manage to review to test and review this PR?

**Comment by DanGarner** [comment]
Hi @dareenzo 

Thank you for your contribution.

Can you explain the actual business use-case(s) that you hope to resolve by providing this functionality?

**Comment by dareenzo** [comment]
Hi @DanGarner 

I want to enforce the UI to use a specific language. 

The way the module works by default is that it picks the language of the browser for the payment pop-up. But in my specific case I'm in Mozambique and most of my users speak only Portuguese  -- though theirs browsers maybe in English. Additionally, at the current stage only Portuguese is available/supported on my site. And having English show on the payment # pop-up is an inconsistency

**Comment by DanGarner** [comment]
Thank you for the explanation - it has helped to understand the context of the contribution that you have made.

We will feed back to Mastercard, so that they can decide about the inclusion of your contribution into the roadmap of the module's development. I can't say when that might happen at this time.

**Comment by abdulmueid** [comment]
Thanks for engaging with us @DanGarner and thanks @dareenzo for the explanation.

Just want to add that this functionality is part of the standard Hosted Checkout / checkout.js provided by Mastercard and I have only exposed it via Magento Admin UI for ease of configuration and use.

https://eu-gateway.mastercard.com/api/documentation/integrationGuidelines/hostedCheckout/integrationModelHostedCheckout.html

Screenshot from 2020-08-12 16-46-15

**Comment by Judihost** [comment]
ihave problem how cn fixed 
look plz links
https://prnt.sc/12y8ni7
https://prnt.sc/12y8r20

ineed help plz

**Comment by dareenzo** [comment]
@Judihost please ask your question at https://marketplace.magento.com/ontap-module-mastercard.html#bv_questions

**Comment by paulophagula** [comment]
@DanGarner @OnTapKristjan did you guys get any feedback form Mastercard on getting this change into your roadmap?

**Comment by chirag-ontap** [comment]
> @DanGarner @OnTapKristjan did you guys get any feedback form Mastercard on getting this change into your roadmap?

@paulophagula for now it is not included in the roadmap. However, we'll update you if any change on this.

File: Model/Ui/Hosted/ConfigProvider.php [hunk 1]
```diff
@@ -59,6 +59,7 @@ public function getConfig()
                 self::METHOD_CODE => [
                     'merchant_username' => $this->config->getMerchantId(),
                     'component_url' => $this->config->getComponentUrl(),
```
Comment (ellenchristine) [inline Model/Ui/Hosted/ConfigProvider.php:61]: Looks like there's an extraneous comma at the end of line 62

**Comment by DanGarner** [comment]
This change might form part of a future revision of the module. However, for now, it is not in the roadmap.

**Pull request closed without merging**